llama-index-integrations/storage/chat_store/llama-index-storage-chat-store-azurecosmosmongovcore/llama_index/storage/chat_store/azurecosmosmongovcore/base.py

```diff
@@ -0,0 +1,157 @@
+import logging
+from abc import ABC
+from typing import List, Optional
+
+from llama_index.core.llms import ChatMessage
+from llama_index.core.storage.chat_store import BaseChatStore
+from pymongo import MongoClient
+from pymongo.database import Database
+from pymongo.collection import Collection
+
+logger = logging.getLogger(__name__)
+
+APP_NAME = "Llama-Index-CDBMongoVCore-ChatStore-Python"
+
+
+# Convert a ChatMessage to a JSON object
+def _message_to_dict(message: ChatMessage) -> dict:
+    return message.dict()
+
+
+# Convert a list of ChatMessages to a list of JSON objects
+def _messages_to_dict(messages: List[ChatMessage]) -> List[dict]:
+    return [_message_to_dict(message) for message in messages]
+
+
+# Convert a JSON object to a ChatMessage
+def _dict_to_message(d: dict) -> ChatMessage:
+    return ChatMessage.model_validate(d)
+
+
+class AzureCosmosMongoVCoreChatStore(BaseChatStore, ABC):
+    """Creates an Azure Cosmos DB NoSql Chat Store."""
+
+    _mongo_client = MongoClient
+    _database = Database
+    _collection = Collection
+
+    def __init__(
+        self,
+        mongo_client: MongoClient,
+        uri: Optional[str] = None,
+        host: Optional[str] = None,
+        port: Optional[int] = None,
+        db_name: Optional[str] = None,
+        collection_name: Optional[str] = None,
+    ):
+        super().__init__(
+            mongo_client=mongo_client,
+            uri=uri,
+            host=host,
+            port=port,
+            db_name=db_name,
+        )
+
+        self._mongo_client = mongo_client
+        self._uri = uri
+        self._host = host
+        self._port = port
+        self._database = self._mongo_client[db_name]
+        self._collection = self._mongo_client[db_name][collection_name]
+
+    @classmethod
+    def from_connection_string(
+        cls,
+        connection_string: str,
+        db_name: Optional[str] = None,
+        collection_name: Optional[str] = None,
+    ):
+        """Creates an instance of AzureCosmosMongoVCoreChatStore using a connection string."""
+        mongo_client = MongoClient(connection_string, appname=APP_NAME)
```

Comment:
May need to do URL encode here if the connection_string contains special characters in password part such as '@'?
If needed, I recommend use `quote_plus` :
`
from urllib.parse import quote_plus
`

Reply:
Updated the code accordingly